### ISSUE TITLE
Viewer fails to initialize inside cross-origin iframe

### ISSUE THREAD

**Issue by themoenen** [epam/miew#524]
**Describe the bug**
The viewer can't be initialized when rendered inside a cross-origin iframe.
This is due to a window.top event listener and is easy to fix.

**To Reproduce**

1. Create a folder `miew-test`

1. Set up two web servers pointing to the same folder, to simulate a cross origin iframe.
   
    In one terminal run:

        python -m http.server 8000
    
    In another terminal run:

        python -m http.server 8001
    
    > Note: Python 3 assumed. For Mac use `python3`

1. Create `viewer.html` with a functioning viewer:

```
<!DOCTYPE html>
<html lang="en">
    <head>
        <link rel="stylesheet" href="https://unpkg.com/miew@0.9.0/dist/Miew.min.css" />
        <script src="https://unpkg.com/@babel/polyfill@7/dist/polyfill.min.js"></script>
        <script src="https://unpkg.com/lodash@4.17.15/lodash.js"></script>
        <script src="https://unpkg.com/three@0.112.1/build/three.min.js"></script>
        <script src="https://unpkg.com/miew@0.9.0/dist/Miew.min.js"></script>
    </head>
    <body>
        <h1>Miew Viewer</h1>
        <div class="miew-container" style="width: 640px; height: 480px"></div>

        <script>
            ;(function () {
                var viewer = new Miew({ load: '1CRN' })
                if (viewer.init()) {
                    viewer.run()
                }
            })()
        </script>
    </body>
</html>
```

1. Create `index.html` with an iframe pointing at the viewer:

```
<!DOCTYPE html>
<html lang="en">
    <head>
    </head>
    <body>
        <iframe src="http://localhost:8001/viewer.html" width="800" height="600"></iframe>
    </body>
</html>
```

1. Open http://localhost:8000

**Expected behavior**
The viewer should initialize successfully.

**Actual behavior**
The viewer doesn't initialize successfully.

**Source of the problem**
The two lines below are responsible. Because we're referring to window.top, this causes a CORS error blocking the further initialization of the viewer. Replacing `window.top` with `window` fixes the problem. If it's important to keep `window.top`, logic should be added to prevent the initialization from crapping out completely.

`Miew.js`

```
window.top.addEventListener('keydown', (event) => {
  self._onKeyDown(event);
});

window.top.addEventListener('keyup', (event) => {
  self._onKeyUp(event);
});

```
If `window.top` can be replaced, I presume it should also be updated in `ObjectControls.js`:

```
ObjectControls.prototype.getKeyBindObject = function () {
  return window.top;
};
```

**Screenshots**
![Screenshot 2024-02-27 at 11 47 26 AM](https://github.com/epam/miew/assets/1200060/1232d58f-949a-485c-943c-8a27c38e2b15)
![Screenshot 2024-02-27 at 11 47 32 AM](https://github.com/epam/miew/assets/1200060/4dd8901a-4365-4f79-982c-7bc7ecadffdf)
![Screenshot 2024-02-27 at 11 48 25 AM](https://github.com/epam/miew/assets/1200060/747d9cb0-c541-4108-9f44-059587cc8dba)

**Context**
 - OS: MacOS Sonoma 14.3.1 (23D60)
 - Browser: Google Chrome 
 - Version: Version 118.0.5993.117 (Official Build) (arm64)
 - npm Miew version: 0.10.1

**Additional context**
If you're ok with replacing `window.top` with `window` I can submit a pull request.

**Comment by paulsmirnov** [comment]
Thank you for reporting the issue. We will investigate it and look for a solution to the problem. Unfortunately, replacing "window.top" with "window" would disable the behavior that we intended to implement originally. We might be ok with deprecating the original intents now but we must carefully consider all consequences.

Which miew version do you use? I'm asking because we don't have plans to make a new release soon. It would be great if you could patch your version to not be blocked by our release schedule. We are kind of limited in resources.

**Comment by themoenen** [comment]
Thanks for the quick reply. I'm on the latest npm version (0.10.1)

I'm already hosting my own modified version, so there's no immediate rush to get this fixed.

I will look into wrapping this into a try/catch as to preserve binding events to the top window and submit a pull request if successfull.

**Comment by paulsmirnov** [comment]
Hopefully, fixed in #550

![shot-20240819-212136](https://github.com/user-attachments/assets/45624775-1f80-4992-ac58-e61d9a780520)

![shot-20240819-212716](https://github.com/user-attachments/assets/2a2f2f17-f3ac-433b-bc00-4c484d8271bb)
